azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -768,46 +772,46 @@ private Props loadJobProps(final ExecutableNode node) throws IOException {
     return props;
   }
 
-  private Props loadFlowPropsFromYamlFile() {
-    final File flowFile = getFlowFile();
-    final Props flowProps = FlowLoaderUtils.getPropsFromYamlFile(this.flow.getId(), flowFile);
-    if (flowFile != null && flowFile.exists()) {
-      flowFile.delete();
+  private Props loadPropsFromYamlFileUtil(final String path) {
+    final File tempDir;
+    Props props = null;
+    try {
+      tempDir = Files.createTempDir();
+    } catch (final IllegalStateException e) {
+      this.logger.error("Failed to create temp directory. " + e);
+      return props;
     }
-    return flowProps;
-  }
-
-  private Props loadJobPropsFromYamlFile(final ExecutableNode node) {
-    final File flowFile = getFlowFile();
-    final String jobPath =
-        node.getParentFlow().getFlowId() + Constants.PATH_DELIMITER + node.getId();
-    final Props props = FlowLoaderUtils.getPropsFromYamlFile(jobPath, flowFile);
-    if (flowFile != null && flowFile.exists()) {
-      flowFile.delete();
+    try {
+      props = FlowLoaderUtils.getPropsFromYamlFile(path, getFlowFile(tempDir));
+    } catch (final Exception e) {
+      this.logger.error("Failed to get props from flow file. " + e);
+    } finally {
+      if (tempDir != null && tempDir.exists()) {
+        try {
+          FileUtils.deleteDirectory(tempDir);
```

Comment:
1). Don't understand why we need to create a temp folder to do these things. 

2). Can we find a way to merge the two try-catch to reduce redundant code?

Reply:
1. We need to create the separate temp directory for each flow file to avoid overwriting the same file by multiple threads concurrently. And the reason why I create temp directory instead of temp file is that the file name will be interpreted as the flow name. And a temp file name is basically some random string combination, so we cannot use.  I've left some comments in azkaban.project.JdbcProjectImpl#getUploadedFlowFile.
2. Do you mean combining createTempDir and getPropsFromYamlFile in the same try-catch block? The reason I'm separating them out is for clear logging and debugging under different exceptions. But you are right they can be combined together, in this way, we will combine the logging message together as well.